smartapps/smartthings/ecobee-connect.src/ecobee-connect.groovy

```diff
@@ -131,20 +188,46 @@ def callback() {
 		def tokenUrl = "https://www.ecobee.com/home/token?${toQueryString(tokenParams)}"
 
 		httpPost(uri: tokenUrl) { resp ->
-			atomicState.refreshToken = resp.data.refresh_token
-			atomicState.authToken = resp.data.access_token
+			state.refreshToken = resp.data.refresh_token
+			state.authToken = resp.data.access_token
 		}
-
-		if (atomicState.authToken) {
+		if ( state.authToken ) {
+			// get jwt for switch+ devices
+			obtainJsonToken()
+		}
+		if (state.authToken) {
```

Comment:
Is there a reason we are doing the two `state.authToken` checks separately?

Reply:
You'll have to ask Todd about that as he added this for the addition of the early access switch API.